src/figuresClasses.ts

```diff
@@ -1,19 +1,69 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(public color: Color,
+    public a: number, public b: number, public c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error("One of the figure's sides is less than or equal to 0");
+    }
 
-export class Circle {
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
```

Comment:
Not fixed previouse comment
`const longestSide = Math.max(a, b, c);`
and use this variable